astropy/io/fits/column.py

```diff
@@ -1429,6 +1429,12 @@ def dtype(self):
                 else:
                     dt = np.dtype((dt.base, dim))
 
+            # On Python 2, force the `name` to `str`
+            # because Numpy structured arrays can't handle unicode `name`
+            # See https://github.com/astropy/astropy/issues/5204
+            if PY2:  # pragma: py2
+                name = str(name)
```

Comment:
@cdeil - Yes, the fix is fine. My fear was that this line was called before the verification, but as it is the opposite it is perfectly fine to assume that the string is ascii-only at this stage.